test/unit/app/utils/test_ussd.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+
+from server import db
+from fixtures.user import UserFactory
+from server.models.ussd import UssdMenu, UssdSession
+
+
+@pytest.fixture(scope='function')
+def menu_display_text_in_lang(test_client, init_database):
+    from server.utils.ussd.ussd import menu_display_text_in_lang
+    return menu_display_text_in_lang
+
+
+@pytest.fixture(scope='function')
+def create_or_update_session(test_client, init_database):
+    from server.utils.ussd.ussd import create_or_update_session
+    return create_or_update_session
+
+
+# TODO(ussd): once preferred_language is added, add another test that sw_KE works and user without preferred_language
+@pytest.mark.parametrize("case,expected", [
+    (1, "foo"),
+    (2, "foo"),
+])
+def test_menu_display_text_in_lang(menu_display_text_in_lang, case, expected):
+    if case == 1:
+        user = None
+    else:
+        # TODO: this is really gross, but I can't use the factory in the parameters since no db initiated... how do?
+        user = UserFactory()
+
+    menu = UssdMenu(display_text_en="foo", display_text_sw="bar")
+    assert menu_display_text_in_lang(menu, user) == expected
+
+
+def test_create_or_update_session(create_or_update_session):
+    # create a session in db
+    session = UssdSession(session_id="1", user_id=2, msisdn="123", ussd_menu_id=3, state="foo", service_code="*123#")
+    db.session.add(session)
+    db.session.commit()
+
+    # test updating existing
+    user = UserFactory(id=2, phone="123")
+    create_or_update_session("1", user, UssdMenu(id=4, name="bar"), "input", "*123#")
+    sessions = UssdSession.query.filter_by(session_id="1")
+    assert sessions.count() == 1
+    session = sessions.first()
+    assert session.state == "bar"
```

Comment:
is there a better way to do comparison of properties? my test failed when I did `assert session == UssdSession(<expected properties>)`

Reply:
I'm pretty sure this is the best way to do this.

`assert session == UssdSession(<expected properties>)` will fail because a new instantiated UssdSession with the same properties of `session` isn't literally the same session (`session` has an id for example).

Also the plus side of asserting properties one by one is that when the test throws an error, we can quickly work out which property failed the assertion